packages/component-library/src/components/plmg-button/plmg-button.scss

```diff
@@ -363,26 +372,94 @@
     }
   }
 
-  // Control: Size
+  // Control: Size & icons
   &.small {
     padding: tokens.$plmg-spacing-x0-5 tokens.$plmg-spacing-x1;
     font-size: tokens.$plmg-font-size-x0-875;
     line-height: tokens.$plmg-font-lineheight-loose;
+
+    &.icon-button {
+      padding: tokens.$plmg-spacing-x1;
+    }
+
+    .icon-left {
+      margin-right: tokens.$plmg-spacing-x0-5;
+    }
+    .icon-right {
+      margin-left: tokens.$plmg-spacing-x0-5;
+    }
+    .icon-center {
+      font-size: centerIconFontSize(
+        tokens.$plmg-font-size-x0-875,
+        tokens.$plmg-font-lineheight-loose
+      );
+    }
   }
   &.medium {
     padding: tokens.$plmg-spacing-x0-75 tokens.$plmg-spacing-x1-5;
     font-size: tokens.$plmg-font-size-x1-125;
     line-height: tokens.$plmg-font-lineheight-loose;
+
+    &.icon-button {
+      padding: tokens.$plmg-spacing-x0-75;
+    }
+
+    .icon-left {
+      margin-right: tokens.$plmg-spacing-x0-75;
+    }
+    .icon-right {
+      margin-left: tokens.$plmg-spacing-x0-75;
+    }
+    .icon-center {
+      font-size: centerIconFontSize(
+        tokens.$plmg-font-size-x1-125,
+        tokens.$plmg-font-lineheight-loose
+      );
+    }
   }
   &.large {
     padding: tokens.$plmg-spacing-x1 tokens.$plmg-spacing-x2;
     font-size: tokens.$plmg-font-size-x1-25;
     line-height: tokens.$plmg-font-lineheight-loose;
+
+    &.icon-button {
+      padding: tokens.$plmg-spacing-x1;
+    }
+
+    .icon-left {
+      margin-right: tokens.$plmg-spacing-x0-75;
+    }
+    .icon-right {
+      margin-left: tokens.$plmg-spacing-x0-75;
+    }
+    .icon-center {
+      font-size: centerIconFontSize(
+        tokens.$plmg-font-size-x1-25,
+        tokens.$plmg-font-lineheight-loose
+      );
+    }
   }
   &.extra-large {
     padding: tokens.$plmg-spacing-x1-5 tokens.$plmg-spacing-x2-5;
     font-size: tokens.$plmg-font-size-x1-5;
     line-height: tokens.$plmg-font-lineheight-normal;
+
+    &.icon-button {
+      padding: tokens.$plmg-spacing-x1;
```

Comment:
The figma shows a padding of:
top: 18px;
right: 18px;
bottom: 20px;
left: 20px;

This feels weird however please confirm with a designer and have them fix it if it is incorrect.